src/box/iproto.cc

```diff
@@ -988,6 +1000,27 @@ iproto_connection_input_buffer(struct iproto_connection *con)
 	return new_ibuf;
 }
 
+static size_t
+iproto_thread_get_max_stream_queue_length(struct iproto_thread *iproto_thread)
+{
+	size_t queue_len_max = 0;
+	struct rlist *entry;
+	rlist_foreach(entry, &iproto_thread->connections) {
+		struct iproto_connection *con =
+			rlist_entry(entry, struct iproto_connection,
+				    in_conn_list);
+		mh_int_t node;
+		mh_foreach(con->streams, node) {
```

Comment:
This may result in performance problems in case the number of streams/connections is huge. Maybe, we could use a historgram for accounting stream queue length? We wouldn't be able to return the exact maximum then, but OTOH we would see an approximate distribution and could get a ballpark figure for the maximum. @alyapunov WDYT?

Reply:
I agree, we should expect lots of concurrent streams. On the other hand we don't need exact max number of streams, an estimation would be good. So I'd prefer O(1) cost here.

Reply:
Thanks Alexander. Then I suggest to drop the last patch for now and implement histogram-based stats in the future. This can be done in a separate PR.

Reply:
Drop last patch